pkg/whip/whip_track_handler.go

```diff
@@ -182,6 +193,12 @@ func (t *whipTrackHandler) processRTPPacket() error {
 				return err
 			}
 
+			t.statsLock.Lock()
+			if t.stats != nil {
+				t.stats.MediaReceived(streamKindFromCodecType(t.remoteTrack.Kind()), int64(len(buf)))
```

Comment:
could MediaReceived block at all? I wasn't sure from this diff if it'll end up making a network request.

In any case, it'd be a good idea to avoid locking while sending this.

Reply:
Currently, this should be safe (no I/O in the call, and while the call takes a lock, there is no real risk of deadlock, or that lock waiting on I/O), not doing the call while holding the lock in indeed safer.